viper.py

```diff
@@ -487,12 +487,15 @@ def fit_chunk(o, chunk, obsname, targ=None, tpltarg=None):
    #     p, e_p = S_mod.fit(x_ok, f_ok, rvs,a, bg, s, c=cc, c0=c0, dx=0.1)
         v = rvs      
 
-    if tplname:
-        p, e_p = S_mod.fit(x_ok, f_ok, v, a, bg, s, t, c=cc , c0=c0, dx=0.1, sig=sig[i_ok])
+    if createtpl:
+        res, rel_fac = False, 1
+    else:
+        res, rel_fac = True, 0
+
+    if tplname or createtpl:
+        p, e_p = S_mod.fit(x_ok, f_ok, v, a, bg, s, t, c=cc, c0=c0, dx=0.1, sig=sig[i_ok], res=res, rel_fac=rel_fac)
```

Comment:
How about
`res=res, rel_fac=rel_fac)`
=>
`res=not createtpl, rel_fac=createtpl)`